Fix router documentation for `druid.router.sql.enable` (#11716)
apache/druid: docs/configuration/index.md

|`druid.router.tierToBrokerMap`|Queries for a certain tier of data are routed to their appropriate Broker. This value should be an ordered JSON map of tiers to Broker names. The priority of Brokers is based on the ordering.|{"_default_tier": "<defaultBrokerServiceName>"}|
|`druid.router.defaultRule`|The default rule for all datasources.|"_default"|
|`druid.router.pollPeriod`|How often to poll for new rules.|PT1M|

> |`druid.router.sql.enable`|Enable routing of SQL queries using strategies. Possible values are `true` and `false`. When set to `true`, the Router uses the provided strategies to determine the broker service for a given SQL query. When set to `false`, the Router uses the `defaultBrokerServiceName`.|`false`|

Reviewer: can you also add what strategies are supported?

Author: The details of strategies are present in docs/design/router.md. I figured we only need a summary here.

> |`druid.router.sql.enable`|Enable routing of SQL queries using strategies. Possible values are `true` and `false`. When set to `true`, the Router uses the provided strategies to determine the broker service for a given SQL query. When set to `false`, the Router uses the `defaultBrokerServiceName`.|`false`|

Reviewer 2: ```suggestion
|`druid.router.sql.enable`|Enable routing of SQL queries using strategies. When`true`, the Router uses the  strategies defined in `druid.router.strategies` to determine the broker service for a given SQL query. When `false`, the Router uses the `defaultBrokerServiceName`.|`false`|
```
No need to enumerate "true/false" for boolean. It's assumed. Suggest clarifying where to configure strategies (`druid.router.strategies`).
